### ISSUE TITLE
the errors should have a better spelling

### ISSUE THREAD

**Issue by weissi** [apple/swift-nio-http2#53]
Currently, the error look like this:

    NIOHTTP2Errors.NoSuchStream(...)

but I think it should be either

    NIOHTTP2Error.noSuchStream(...) // imitating an open enum on top of a struct

or

    NIOHTTP2Errors.NoSuchStreamError(...)

IMHO, preferably the former

**Comment by natikgadzhi** [comment]
If we do that, though, we should deprecate the existing ones, and leave them there for another release, right?

**Comment by Lukasa** [comment]
Yeah, but this is non-urgent enough that I’d be inclined to leave these until we did a new major release.
